sqlmesh/core/engine_adapter/base.py

```diff
@@ -366,22 +398,20 @@ def create_view(
             create_kwargs: Additional kwargs to pass into the Create expression
         """
         schema: t.Optional[exp.Table | exp.Schema] = exp.to_table(view_name)
-
-        if isinstance(query_or_df, DF_TYPES):
-            if PySparkDataFrame is not None and isinstance(query_or_df, PySparkDataFrame):
-                query_or_df = query_or_df.toPandas()
-
-            if not isinstance(query_or_df, pd.DataFrame):
-                raise SQLMeshError("Can only create views with pandas dataframes.")
-
+        df = self.try_get_pandas_df(query_or_df)
+        pyspark_df = self.try_get_pyspark_df(query_or_df)
+        if df is not None or pyspark_df:
+            if pyspark_df:
+                df = pyspark_df.toPandas()
```

Comment:
```
df = df or pyspark_df.toPandas()
```
maybe

Reply:
Great idea!

Reply:
So we talked and the downside would be that the try_get would then have a cost in the case that the value was a pyspark df. Therefore you should consider that in implementation. With the current approach these try_gets are free so we decided to stick with the current approach.